src/monster.cpp

```diff
@@ -1024,11 +1016,41 @@ void Monster::onThinkYell(uint32_t interval)
 	}
 }
 
+bool Monster::walkToSpawn()
+{
+	if (walkingToSpawn || !spawn || !targetList.empty()) {
+		return false;
+	}
+
+	int32_t distance = std::max(Position::getDistanceX(position, masterPos), Position::getDistanceY(position, masterPos));
+	if (distance == 0) {
+		return false;
+	}
+
+	std::vector<Direction> dirList;
+	if (!getPathTo(masterPos, dirList, 0, std::max<int32_t>(0, distance - 5), true, true, distance)) {
+		return false;
+	}
+
+	walkingToSpawn = true;
+	startAutoWalk(dirList);
```

Comment:
don't use a temp vector if you have access to the object's private data
example:
```c++
listWalkDir.clear();
if (getPathTo(masterPos, listWalkDir, 0, 0, true, true)) {
	startAutoWalk();
}
```

if in the future you want to use getPathTo from a place where you don't have access to a creature's private data, then use the temporary vector, but don't forget to move it correctly with `std::move()`

This is an optimization that was done a few months ago, please use it correctly.